### PR TITLE
fix bug helm repo paging query

### LINKED ISSUE #5181: HelmRepo paging query error
**Describe the Bug**

You don't need to remove this comment section, it's invisible on the issues page.

**When the added HelmRepo is more than 10, the paging query error.**

**Versions Used**
KubeSphere: v3.2.1
Kubernetes: (If KubeSphere installer used, you can skip this)

image

image

### PR DESCRIPTION
**Pull request by yongxingMa**
### What type of PR is this?
/kind bug

### Which issue(s) this PR fixes:

Fixes #5181 

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
fix bug helm repo paging query
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/models/openpitrix/repos.go [hunk 1]
```diff
@@ -298,8 +298,6 @@ func (c *repoOperator) ListRepos(conditions *params.Conditions, orderBy string,
 	}
 
 	totalCount := len(repos)
-	start, end := (&query.Pagination{Limit: limit, Offset: offset}).GetValidPagination(totalCount)
-	repos = repos[start:end]
 	items := make([]interface{}, 0, len(repos))
 	for i, j := offset, 0; i < len(repos) && j < limit; i, j = i+1, j+1 {
```
Comment (mangoGoForward) [inline pkg/models/openpitrix/repos.go:302]: I think we should not to remove those line, but should fix the for loop.
when we want to get the second page, the `start` is 10, `end` is 11, `len(repos)` is 1 and `i` is 10, so if `i < len(repos)` is false

**Review by mangoGoForward — CHANGES_REQUESTED** [review]

File: pkg/models/openpitrix/repos.go (hunk 1 above)
Comment (xyz-li) [reply pkg/models/openpitrix/repos.go:302]: How about this?
```
	start, end := (&query.Pagination{Limit: limit, Offset: offset}).GetValidPagination(totalCount)
	repos = repos[start:end]
	items := make([]interface{}, 0, len(repos))
	for i := range repos {
		items = append(items, convertRepo(repos[i]))
	}
```

Comment (mangoGoForward) [reply pkg/models/openpitrix/repos.go:302]: more better

Comment (yongxingMa) [reply pkg/models/openpitrix/repos.go:302]: > How about this?
> 
> ```
> 	start, end := (&query.Pagination{Limit: limit, Offset: offset}).GetValidPagination(totalCount)
> 	repos = repos[start:end]
> 	items := make([]interface{}, 0, len(repos))
> 	for i := range repos {
> 		items = append(items, convertRepo(repos[i]))
> 	}
> ```

Well, this way.

**Pull request merged**